private/tools/java/com/github/bazelbuild/rules_jvm_external/Coordinates.java

```diff
@@ -36,29 +36,64 @@ public Coordinates(String coordinates) {
           "Bad artifact coordinates "
               + coordinates
               + ", expected format is"
-              + " <groupId>:<artifactId>[:<extension>[:<classifier>][:<version>]");
+              + " <groupId>:<artifactId>[:<version>][:<classifier>][@<extension>");
     }
 
     groupId = Objects.requireNonNull(parts[0]);
     artifactId = Objects.requireNonNull(parts[1]);
 
+    boolean isGradle =
```

Comment:
Please document this constructor formally on what formats are accepted. It's getting a bit harder to understand this code, and AFAIUC, we are trying to use this to canonicalize the format, so some documentation will be useful.

Reply:
Done. I have added a little white-lie about how we handle `a:b:c`, but I'll create a follow up PR to make that true.